### PR TITLE
Prep for 0.12.2 release

### PR DESCRIPTION
**Pull request by j9ac9k**
PR here includes a changelog of the changes made to pyqtgraph since 0.12.2.  I left out some PRs that were merged off the list (things like static code checker cleanups, documentation expansion, moving the test data into the repository, things of that nature).  I'm not opposed to adding them if someone thinks they should be added; but I was trying to keep things a bit more general; git-blame is good for finite details 😆 

Thank you to all the contributors and maintainers that went into this release, while a patch version, there are _substantial_ improvements to the library here that wouldn't have been possible without the work of @pijyoi @outofculture @NilsNemitz and others that have volunteered your time/effort/skills to make this library better for everyone. 

Also a special thank you/apology to @chrismullins for making some changes to a PR they created 4 years earlier (we'll do our best to not let that happen again!).

EDIT:   A few issues popped up that we should address before doing the release

* [x] #1847 
* [x] #1843
* [x] #1857
* [x] #1862
* [x] #1846

### CODE REVIEW

**Comment by pijyoi** [comment]
Maybe mention numba on README.md

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+    - Qt6 6.0 support has been removed in favor of Qt6 6.1+
+    - More numba utilization, specifically for makeARGB 
+    - Substantial ImageItem performance improvements have been made thanks to @pijyoi and @outofculture
+    - Significant performance improvements made to ScatterPlotItem and LinePlots
+    - More Expansive ColorMap features/support (more are coming!)
+  
+  New Features
+    - #1318 Added TargetItem
+    - #1707 Added Qt 6.1 support
+    - #1729 Allow gradient position to be configured on a historgram
```
Comment (NilsNemitz) [inline CHANGELOG:13]: histogram

Comment (j9ac9k) [reply CHANGELOG:13]: thank you!

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,51 @@
+pyqtgraph-0.12.2
+
+  Highlights
+    - Qt6 6.0 support has been removed in favor of Qt6 6.1+
+    - More numba utilization, specifically for makeARGB 
+    - Substantial ImageItem performance improvements have been made thanks to @pijyoi and @outofculture
+    - Significant performance improvements made to ScatterPlotItem and LinePlots
+    - More Expansive ColorMap features/support (more are coming!)
```
Comment (NilsNemitz) [inline CHANGELOG:8]: remove expansive?   Otherwise, don't capitalize it.

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
+    - #1729 Allow gradient position to be configured on a histogram
+    - #1742 Better support for plotting with gradients
+    - #1776 Add GLTextItem
+    - #1797 Add ColorMap linearization (using CIELab calculations), colorDistance functionality
+
+  Performance enhancement:
+    - #1738, #1695, 1786, #1768, 1794 - ImageItem/makeARGB performance improvements
+    - #1724 Use math module for scalar values math instead of numpy functions
+    - #1796 Greatly speed up line plots with use-case of connect='all'
+    - #1817 Speedup some cases of connect='finite' (few discontinuities)
```
Comment (NilsNemitz) [inline CHANGELOG:22]: Speed up

File: CHANGELOG [hunk 4]
```diff
@@ -1,3 +1,51 @@
...
+    - #1833 Use QPainter.drawPixmapFragments
```
Comment (NilsNemitz) [inline CHANGELOG:23]: ...in ScatterPlotItem   
??

File: CHANGELOG [hunk 5]
```diff
@@ -1,3 +1,51 @@
...
+
+  Bug Fixes:
+    - #1700 Fixed ROI getArrayRegion
+    - #1748 Fixed bug when plotting boolean arrays in PlotDataItem
+    - #1791 Callable LUTs being used on the ImageItem substrates
+    - #1783 Fix memory leak in GLMeshItem
+    - #1802 Updated cx_freeze example and added workaround for template files
+    - #1804 Fix mouseClick handling for Qt6 on ROIs
+    - #1799 Force cameraPosition() to return a Vector in GLViewWidget
+    - #1809 Sanitize ShowGrid Alpha Slider Input PlotItem template
```
Comment (NilsNemitz) [inline CHANGELOG:33]: "Sanitze PlotItem showGrid parameters"
This didn't touch the template. Or the slider.

File: CHANGELOG [hunk 6]
```diff
@@ -1,3 +1,51 @@
...
+
+  API/Behavior Changes:
+    - #519 Expose clickable property in PlotDataItem
+    - #1772 Keep ColorMap values for RGBA as uint8
+    - #1736 RemoteGraphicsView is now hidpi aware
+    - #1779 Have SpinBox use fallback minStep in dec mode
+    - #1706 Colors defined with hex string values must start with a #
+    - #1819 Added method to disable autoscaling for HistogramLUTItem
+    - #1638 Expose number of subsamples in ImageItem auto-level determination
+    - #1824 Remove little-endian assumption for image export
```
Comment (NilsNemitz) [inline CHANGELOG:46]: I would add a mention that all the tests are in a different place...

**Comment by j9ac9k** [comment]
> Maybe mention numba on README.md

completely forgot it wasn't on the readme!  thanks for reminding, added.

File: CHANGELOG (hunk 2 above)
Comment (j9ac9k) [reply CHANGELOG]: done!

**Review by NilsNemitz — APPROVED** [review]

**Comment by j9ac9k** [comment]
Putting a hold on this until we resolve #1843

**Comment by pijyoi** [comment]
Could we also add the following fix for 0.12.2? https://github.com/pyqtgraph/pyqtgraph/pull/1871/commits/e76328ab0eff1c58577bbf62a38c97ffa34b4e02
If preferable, I can split it to its own PR.

**Comment by j9ac9k** [comment]
> Could we also add the following fix for 0.12.2? [e76328a](https://github.com/pyqtgraph/pyqtgraph/commit/e76328ab0eff1c58577bbf62a38c97ffa34b4e02)
> If preferable, I can split it to its own PR.

 You can submit the PR, but when I hop on a machine with some time to spare, if I don't see a PR from you, I'll create one and cherry-pick your commit if that works for you.

**Comment by ntjess** [comment]
I've also found that #1873 makes a surprising difference in quick usability, if you're OK with adding that short PR as well.

**Review by ksunden — APPROVED** [review]

**Comment by j9ac9k** [comment]
oh right, we merged some more PRs, I should probably update the changelog... I'll do that right now

**Comment by j9ac9k** [comment]
ok, I think I got everything that should be on the changelog, I didn't list every PR we merged but tried to keep it to items that were a bit higher level and more user-facing in some way.

**Pull request merged**